File: 1680-concatenation-of-consecutive-binary-numbers/1680-concatenation-of-consecutive-binary-numbers.py
class Solution:
    def concatenatedBinary(self, n: int) -> int:
        M = 10 ** 9 + 7
        l, ans = 0, 0
        
        for i in range(1, n + 1):
            if i & (i - 1) == 0:
                l += 1
            ans = ((ans << l) | i) % M
        return ans
            
        

File: 1680-concatenation-of-consecutive-binary-numbers/1680-concatenation-of-consecutive-binary-numbers.go
func concatenatedBinary(n int) int {
    ans, l, M := 0, 0, 1_000_000_007
    for i := 1; i <= n; i++ {
        if (i & (i - 1) == 0){
            l += 1
        }
        ans = ((ans << l) | i) % M
    }
    return ans
}